### PR TITLE
Added integration test for single sharded DF job

### PR DESCRIPTION
**Pull request by shreyakhajanchi**
PR for merging sharded jobs - [link](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1846/)

Will merge IT PR after the above PR is merged.

[This](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1844) is the draft PR with all changes combined and all tests passing.

### CODE REVIEW

File: v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerITBase.java [hunk 1]
```diff
@@ -198,10 +205,6 @@ protected LaunchInfo launchDataflowJob(
         new HashMap<>() {
           {
             put("inputFilePattern", getGcsPath(gcsPathPrefix + "/cdc/"));
-            put(
-                "streamName",
-                String.format(
-                    "projects/%s/locations/us-central1/streams/test-stream-name", PROJECT));
```
Comment (darshan-sj) [inline v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerITBase.java:204]: streamName is optional now?

File: v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DatastreamToSpannerSingleDFShardedMigrationIT.java [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+      }
+    }
+  }
+
+  /**
+   * Cleanup dataflow job and all the resources and resource managers.
+   *
+   * @throws IOException
+   */
+  //  @AfterClass
```
Comment (darshan-sj) [inline v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DatastreamToSpannerSingleDFShardedMigrationIT.java:119]: Is this commented out intentionally?

File: v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DatastreamToSpannerSingleDFShardedMigrationIT.java [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+  //  public static void cleanUp() throws IOException {
+  //    for (DatastreamToSpannerSingleDFShardedMigrationIT instance : testInstances) {
+  //      instance.tearDownBase();
+  //    }
+  //    ResourceManagerUtils.cleanResources(spannerResourceManager, pubsubResourceManager);
+  //  }
+
+  @Test
+  public void multiShardMigration() {
+    // Two dataflow jobs are running corresponding to two physical shards containing two logical
```
Comment (darshan-sj) [inline v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DatastreamToSpannerSingleDFShardedMigrationIT.java:129]: Stale comment?

Comment (darshan-sj) [reply v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DatastreamToSpannerSingleDFShardedMigrationIT.java:129]: Can you update the comment describing how this is testing multiple shards migration with single dataflow job? Is it users responsibility to configure the notification for all the shards to the same Pubsub topic?

**Review by darshan-sj — APPROVED** [review]
Looks good overall

File: v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerITBase.java (hunk 1 above)
Comment (shreyakhajanchi) [reply v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerITBase.java:204]: yes, its an optional parameter

File: v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DatastreamToSpannerSingleDFShardedMigrationIT.java (hunk 3 above)
Comment (shreyakhajanchi) [reply v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DatastreamToSpannerSingleDFShardedMigrationIT.java]: done

**Review by darshan-sj — APPROVED** [review]
LGTM

**Pull request merged**